src/bot.js

```diff
@@ -1,38 +1,186 @@
+ * @returns {Promise<NASAData>}
  */
-async function main() {
-  const client = new djs.Client();
-  
-  client.on('ready', onReady.bind(client));
-  client.on('message', onMessage.bind(client));
+async function getAPOD() {
+  return new Promise((res, rej) => {
+    let result = '';
+    const req = https.get(NASA_URL, (response) => {
+      // Listen for incoming data on the stream
+      response.on('data', data => result += data);
+      // Handle errors on the stream and call the Promise's reject function. 
+      // This let's users of this function know that something went wrong
+      response.on('error', rej);
+      // When the stream has finished sending data let's send it back to the
+      // original function all
+      response.on('end', _ => {
+        // deserialize the data (turn a buffer into object data that our code
+        // can understand)
+        if (result.length === 0) {
+          const msg = "Result was empty and status code was: "
+            + response.statusCode;
+          rej(new Error(msg));
+        }
+        const deserialized = JSON.parse(result);
+        // call the Promise's resolve callback function which let the user of
+        // the function know that this is what they want
+        res(deserialized);
+      });
+    });
+  });
+}
 
-  await client.login(TOKENS.discord);
+/**
+ * @param {string} jobChannel
+ * This function sends the APOD as a formatted embed Discord message
+ */
+async function sendAPOD(jobChannel) {
+  const data = await getAPOD();
+  const channel = await client.channels.fetch(jobChannel, true);
+  console.debug(data);
 
-  schedule.scheduleJob('0 8 * * *', async () => {
-    const data = await getAPOD();
-    const channel = await client.channels.fetch(CHANNEL, true);
-    console.debug(data);
+  const embed = new discordjs.MessageEmbed();
 
-    const embed = new djs.MessageEmbed();
+  if (data.media_type === 'image') {
+    if (data.hdurl) embed.setImage(data.hdurl);
+    else embed.setImage(data.url);
+    embed.setTitle(data.title);
+  }
+  else if (data.url.includes('youtube')) {
+    let link = 'https://www.youtube.com/watch?v=';
+    link += data.url.substring(30);
+    await channel.send(link).catch(console.log);
+  }
+  else await channel.send(data.url).catch(console.log);
 
-    embed.setTitle(data.title)
-         .setImage(data.hdurl)
-         .setDescription(data.explanation)
-         .setColor('PURPLE')
-         .setFooter('APOD for ' + data.date + ' • Photo by ' + data.copyright);
+  let footer = 'APOD for ' + data.date;
+  if (data.copyright) footer += ` • ${data.media_type} by ` + data.copyright;
 
-    await channel.send(embed).catch(console.log);
-  });
+  embed
+    .setDescription(data.explanation)
+    .setColor('PURPLE')
+    .setFooter(footer);
+
+  await channel.send(embed).catch(console.log);
+}
+
+/**
+ * This function is called when a connection between the bot and Discord's
+ * websocket has been established.
+ */
```

Comment:
tip: Although this JSDoc doesn't disclose the return type it doesn't need to since functions starting with "on" are already treated as event handlers which event handlers never return data unless the [EventEmitter](https://nodejs.org/api/events.html#events_class_eventemitter) calling this callback function is expecting data back (good for command frameworks to tell if a command failed or not)

Reply:
👍